### PR TITLE
Fix REST urls for WMTS layers

### LINKED ISSUE #208: WMTS REST don't seem to work ...
With a WMTS REST I have the following error messages:
`ERROR o.m.print.map.tiled.TileLoaderTask - Error making tile request: http://ows.asitvd.ch/http://ows.asitvd.ch/wmts/`

The 'http://ows.asitvd.ch/' URL part is duplicated...

The layer definition I have:
```
{
    "baseURL": "http://ows.asitvd.ch/wmts/",
    "opacity": 1,
    "type": "wmts",
    "layer": "asitvd.fond_couleur",
    "version": "1.0.0",
    "requestEncoding": "REST",
    "style": "default",
    "dimensions": ["DIM1", "ELEVATION"],
    "dimensionParams": {
        "DIM1": "default",
        "ELEVATION": "0"
    },
    "matrixSet": "21781",
    "imageFormat": "image/png",
    "matrices": [{
        "identifier": 0,
        "matrixSize": [38, 25],
        "scaleDenominator": 178571.42857142858,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 1,
        "matrixSize": [94, 63],
        "scaleDenominator": 71428.57142857143,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 2,
        "matrixSize": [188, 125],
        "scaleDenominator": 35714.28571428572,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 3,
        "matrixSize": [375, 250],
        "scaleDenominator": 17857.14285714286,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 4,
        "matrixSize": [750, 500],
        "scaleDenominator": 8928.57142857143,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 5,
        "matrixSize": [938, 625],
        "scaleDenominator": 7142.857142857143,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 6,
        "matrixSize": [1875, 1250],
        "scaleDenominator": 3571.4285714285716,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 7,
        "matrixSize": [3750, 2500],
        "scaleDenominator": 1785.7142857142858,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 8,
        "matrixSize": [7500, 5000],
        "scaleDenominator": 892.8571428571429,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 9,
        "matrixSize": [18750, 12500],
        "scaleDenominator": 357.14285714285717,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }, {
        "identifier": 10,
        "matrixSize": [37500, 25000],
        "scaleDenominator": 178.57142857142858,
        "tileSize": [256, 256],
        "topLeftCorner": [420000, 350000]
    }]
}
```

### PR DESCRIPTION
**Pull request by tsauerwein**
Closes #208

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/map/tiled/wmts/WMTSLayer.java [hunk 1]
```diff
@@ -185,4 +167,31 @@ public Double getLayerDpi() {
...
+                              final WMTSLayerParam layerParam) throws URISyntaxException {
+        String path = layerParam.baseURL;
+        if (layerParam.dimensions != null) {
+            for (int i = 0; i < layerParam.dimensions.length; i++) {
+                String dimension = layerParam.dimensions[i];
+                final String value = layerParam.dimensionParams.getString(dimension.toUpperCase());
+                path = path.replace("{" + dimension + "}", value);
+            }
+        }
+        path = path.replace("{TileMatrixSet}", layerParam.matrixSet);
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/map/tiled/wmts/WMTSLayer.java]: The `{style}` (without capital) is missing ...
See table 32 in the WMTS specifications.

Comment (tsauerwein) [inline core/src/main/java/org/mapfish/print/map/tiled/wmts/WMTSLayer.java]: Added with  1772958.

**Comment by sbrunner** [comment]
Looks good for me :-)

**Comment by tsauerwein** [comment]
Thanks for the review.

**Pull request merged**